flytekit/core/type_engine.py

```diff
@@ -54,7 +55,14 @@
     Union,
     Void,
 )
-from flytekit.models.types import LiteralType, SimpleType, StructuredDatasetType, TypeStructure, UnionType
+from flytekit.models.types import (
+    LiteralType,
+    SimpleType,
+    StructuredDatasetType,
+    TypeStructure,
+    UnionType,
+)
+from flytekit.types.pickle import FlytePickle
```

Comment:
I think this is a circular import. We might need to import it inside `dict_to_generic_literal()`